discovery/gossiper.go

```diff
@@ -423,8 +424,9 @@ type AuthenticatedGossiper struct {
 	// height specified in the future. We will save them and resend it to
 	// the chan networkMsgs once the block height has reached. The cached
 	// map format is,
-	//   {blockHeight: [msg1, msg2, ...], ...}
-	futureMsgs *lru.Cache[uint32, *cachedNetworkMsg]
+	//   {msgID1: msg1, msgID2: msg2, ...}
+	futureMsgs  *lru.Cache[uint64, *cachedFutureMsg]
```

Comment:
Do we also need to do this to the premature message cache?

Reply:
Good question. For premature messages, it's a bit different as we only expect premature `ChannelUpdate`s from the remote peer iff we are a participant of the channel, which means this `cache` could not grow that big? Anyway, have a local PR to enforce the logic "only allow premature `ChannelUpdate` iff we are a participant", and with another PR that fixes the validation barrier, this `cache` should then not be an issue.